### PR TITLE
Applied code checks

### PR DESCRIPTION
**Pull request by photoniker**
I have applied checks and fixes onto the source code using pre_commit hooks:
- pyupgrade
- codespell
- fix-byte-order-marker

### CODE REVIEW

**Review by photoniker — COMMENTED** [review]
I didn't find something

File: docs/docsTmpl/sphinxext/docscrape.py [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+"""Extract reference documentation from the NumPy source tree."""
 
 import inspect
 import textwrap
 import re
 import pydoc
 from io import StringIO
-#from StringIO import StringIO
+
+# from StringIO import StringIO
```
Comment (magro11) [inline docs/docsTmpl/sphinxext/docscrape.py:9]: remove this line

File: itom-packages/apps/roughnessEvaluator/profile_roughness.py [hunk 2]
```diff
@@ -63,7 +62,7 @@ def loadItem(self, dataObj: dataObject):
         ):
             ui.msgWarning(
                 "wrong unit",
-                "Value and horizontal axis unit must be nm, �m or mm",
+                "Value and horizontal axis unit must be nm, �m or mm",
```
Comment (magro11) [inline itom-packages/apps/roughnessEvaluator/profile_roughness.py:65]: the special character should be µm, if the file is stored as utf8. Please check this (e.g. with notepad++). Check encoding and that the µ-sign is correct.

File: itom-packages/devices/Axiovert200M/simpleGUI_ui.py
Comment (magro11) [inline itom-packages/devices/Axiovert200M/simpleGUI_ui.py:1]: remove this file and make sure, that PyQt6 uic is not compiling any ui-files to python files

File: itom-packages/devices/thorlabsLabJack/MLJ050.py [hunk 3]
```diff
@@ -93,8 +93,8 @@ def __init__(self):
         # header (13 04 0e 00 d0 01)
         # channel ident (01 00)
         # MinVelocity (00 00 00 00) = 0
-        # Acceleration (36 a6 01 00) = 108086 -> *90.9 wegen komischem Faktor = 9825017.4 -> ca. 7.99 mm/sec/sec (durch 1228800 microsteps)
-        # Max Velocity (e0 52 cb 0b) = 197874400 -> /53.68 wegen komischem Faktor = 3686184.50 -> ca. 3mm/sec (durch 1228800 microsteps)
+        # Acceleration (36 a6 01 00) = 108086 -> *90.9 wegen komischem Factor = 9825017.4 -> ca. 7.99 mm/sec/sec (durch 1228800 microsteps)
```
Comment (magro11) [inline itom-packages/devices/thorlabsLabJack/MLJ050.py:96]: translate this line and the next one to english

File: itom-packages/itomAlgorithmsStubsGen.py [hunk 4]
```diff
@@ -99,15 +97,12 @@ def generateAlgorithmStubs():
         if algo.isidentifier():
             algoItems.append(parseAlgorithmString(algos[algo]))
 
-    header = (
-        """# coding=iso-8859-15
+    header = """# coding=iso-8859-15
```
Comment (magro11) [inline itom-packages/itomAlgorithmsStubsGen.py:100]: I would say, that this ```coding=iso-...``` line can be removed. Remove the string ```# coding=iso-8859-15```

**Review by magro11 — CHANGES_REQUESTED** [review]
Please check my inline-comments. Especially make sure, that gtest is not changed and set to exclude list and make sure, that PyQT6-auto-generated ui.py files will not be generated any more and will be removed.

File: .codespell_ignore_words.txt [hunk 5]
```diff
@@ -0,0 +1,18 @@
+afile
+enver
+frustrum
+GL
+inOut
+inout
+ist
+manuel
+marge
```
[the remark this answers is not in the document]
Comment (photoniker) [reply .codespell_ignore_words.txt:9]: Manuel is an author named in some file. Marge is from Simpsons used in an example

File: Qitom/translation/qitom_zh.ts [hunk 6]
```diff
@@ -8860,7 +8860,7 @@ or correct the base directory in the preferences dialog of itom.</source>
     <name>ito::WidgetPropEditorAPI</name>
     <message>
         <location filename="../ui/widgetPropEditorAPI.cpp" line="+49"/>
-        <source>Base path for relative pathes: </source>
```
[the remark this answers is not in the document]
Comment (photoniker) [reply Qitom/translation/qitom_zh.ts:8863]: But still pathes is wrong?!

**Comment by photoniker** [comment]
For python unittest I see such fails: 

- dataobject_np_conversion.py:
`======================================================================
FAIL: test_continuousDataObject2NpArray 
... (dataobject_np_conversion.DataObjectNpConversion.test_continuousDataObject2NpArray)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\itom\sources\itom\python_unittests\dataobject_np_conversion.py", line 24, in 
... test_continuousDataObject2NpArray
    self.assertIs(b1.base, a)
AssertionError: (dataObject('uint8', [2 x 3 x 3], continuous: 1, owndata: 1), <capsule object NULL at 
... 0x000002D908301B00>) is not dataObject('uint8', [2 x 3 x 3], continuous: 1, owndata: 1)`

- itom_jedilib.py:
`======================================================================
FAIL: test_help_notypehints (itom_jedilib.ItomJediLibTest.test_help_notypehints)
.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\itom\sources\itom\python_unittests\itom_jedilib.py", line 229, in test_help_notypehints
    self.assertEqual(len(h), 0)
AssertionError: 1 != 0`

**Comment by BBertschinger** [comment]
Do we have a documentation on how to apply our pre-commit hooks?

**Comment by photoniker** [comment]
@BBertschinger see into the README of https://github.com/itom-project
image

**Pull request merged**